arcgis-ios-sdk-samples/Maps/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningViewController.swift

```diff
@@ -0,0 +1,306 @@
+    /// The label to display sensors info.
+    @IBOutlet var sensorStatusLabel: UILabel!
+    
+    // MARK: Properties
+    
+    /// The measurement formatter for sensor accuracy.
+    let measurementFormatter: MeasurementFormatter = {
+        let formatter = MeasurementFormatter()
+        formatter.unitStyle = .short
+        formatter.unitOptions = .providedUnit
+        return formatter
+    }()
+    
+    /// The app-wide API key.
+    let apiKey = AGSArcGISRuntimeEnvironment.apiKey
+    /// A indoors location data source based on sensor data, including but not
+    /// limited to radio, GPS, motion sensors.
+    var indoorsLocationDataSource: AGSIndoorsLocationDataSource?
+    
+    /// The current floor level reported by the indoors location data source.
+    var currentFloor: Int! {
+        willSet(newFloor) {
+            if newFloor != currentFloor {
+                displayFeatures(onFloor: newFloor)
+            }
+        }
+    }
+    
+    // MARK: Methods
+    
+    /// Load an IPS-enabled web map from a portal.
+    func makeMap() -> AGSMap {
+        // Temporarily unset the API key for this sample.
+        AGSArcGISRuntimeEnvironment.apiKey = ""
+        
+        // The portal that hosts the IPS-enabled map.
+        let portal = AGSPortal(url: URL(string: "https://viennardc.maps.arcgis.com")!, loginRequired: true)
+        // WARNING: Never hardcode login information in a production application.
+        // This is done solely for the sake of the sample.
+        let credential = AGSCredential(user: "tester_viennardc", password: "password.testing12345")
+        portal.credential = credential
+        // A floor-aware, IPS-enabled web map for floors of Esri Building L in Redlands.
+        let map = AGSMap(item: AGSPortalItem(portal: portal, itemID: "89f88764c29b48218366855d7717d266"))
+        map.load { [weak self] error in
+            if let error = error {
+                self?.presentAlert(error: error)
+            } else {
+                self?.findPositioningTable(map: map)
+            }
+        }
+        return map
+    }
+    
+    /// Find the IPS positioning table to set up the location data source.
+    func findPositioningTable(map: AGSMap) {
+        let tables = map.tables as! [AGSServiceFeatureTable]
+        loadTables(tables) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .failure:
+                self.presentAlert(error: SetupError.failedToLoadFeatureTables)
+            case .success:
+                if let featureTable = tables.first(where: { $0.tableName == "ips_positioning" }) {
+                    self.setupIndoorsLocationDataSource(positioningTable: featureTable)
+                } else {
+                    self.presentAlert(error: SetupError.positioningTableNotFound)
+                }
+            }
+        }
+    }
+    
+    /// Set up indoors location data source using IPS positioning table.
```

Comment:
The comment here has room for improvement, but I'll change after other feedback.